File: samples/java_layout/src/main/java/io/customer/android/sample/java_layout/ui/inline/SingleInlineFragment.java
package io.customer.android.sample.java_layout.ui.inline;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import io.customer.android.sample.java_layout.databinding.FragmentSingleInlineBinding;
import io.customer.android.sample.java_layout.ui.core.BaseFragment;

public class SingleInlineFragment extends BaseFragment<FragmentSingleInlineBinding> {
    
    private static final String TAG = "SingleInlineFragment";

    @NonNull
    @Override
    protected FragmentSingleInlineBinding inflateViewBinding() {
        return FragmentSingleInlineBinding.inflate(getLayoutInflater());
    }

    @Override
    protected void setupContent() {
        // Set action listener for the inline in-app message
        binding.inappMessage.setActionListener(new InlineMessageActionListenerImpl(requireContext(), "Single Tab"));
    }

    public static SingleInlineFragment newInstance() {
        return new SingleInlineFragment();
    }
}

File: samples/kotlin_compose/src/main/java/io/customer/android/sample/kotlin_compose/ui/inline/components/SingleInlineFragment.kt
package io.customer.android.sample.kotlin_compose.ui.inline.components

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import io.customer.android.sample.kotlin_compose.databinding.FragmentSingleInlineBinding

class SingleInlineFragment : Fragment() {
    private var _binding: FragmentSingleInlineBinding? = null
    private val binding get() = _binding!!

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        _binding = FragmentSingleInlineBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onDestroyView() {
        super.onDestroyView()
        _binding = null
    }

    companion object {
        private const val TAG = "SingleInlineFragment"

        @JvmStatic
        fun newInstance() = SingleInlineFragment()
    }
}
